### PR TITLE
add factory pattern for minio

### PR DESCRIPTION
**Pull request by jackschofield23**
Signed-off-by: Jack Schofield <jack.schofield@answerdigital.com>

### CODE REVIEW

File: src/Storage/MinIo/IMinIoClientFactory.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (mocsharp) [inline src/Storage/MinIo/IMinIoClientFactory.cs:1]: missing copyright

File: src/Storage/MinIo/IMinIoClientFactory.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+using System.Collections.Generic;
+using System.Text;
+using Minio;
+
+namespace Monai.Deploy.Storage.MinIo
+{
+    public interface IMinIoClientFactory
```
Comment (mocsharp) [inline src/Storage/MinIo/IMinIoClientFactory.cs:8]: Is the need of adding a factory here for testing or something else, just curious.

Comment (jackschofield23) [reply src/Storage/MinIo/IMinIoClientFactory.cs:11]: When Joe is trying to set up the integration tests, the minioclient creation is failing. This should allow us to abstract the IMinIoClientFactory in the test framework and return our own MinIoClient

Comment (mocsharp) [reply src/Storage/MinIo/IMinIoClientFactory.cs:11]: Just FYI, the other option is to mock HttpClient but it's fine to use a factory as well, which ever is easier.

**Review by mocsharp — APPROVED** [review]

**Pull request merged**